EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -112,7 +113,8 @@ V2::World::World(const EU4::World& sourceWorld,
 	Log(LogLevel::Progress) << "65 %";
 
 	LOG(LogLevel::Info) << "-> Merging Nations";
-	addUnions();
+	addUnions(sourceWorld.decentralizedHRE(), findEmperorHRE()->getPrimaryCulture());
+	decentralizeHRE(sourceWorld.decentralizedHRE(), findEmperorHRE()->getPrimaryCulture());
```

Comment:
decentralizeHRE drops the emperor's union tag, but you do this *after* applying them in addUnions? Reverse these functions.

Reply:
Also, both of these functions will crash in the event eu4 had an emperor but he didn't survive conversion process as you're addressing findEmperorHRE()->getPrimaryCulture() without checking if  findEmperorHRE() returned nullptr.

Reply:
addUnions is doing core adding, while the other function removes union from cultures.txt

Reply:
I'm aware, which is why they need to be reversed.

Reply:
> Also, both of these functions will crash in the event eu4 had an emperor but he didn't survive conversion process as you're addressing findEmperorHRE()->getPrimaryCulture() without checking if findEmperorHRE() returned nullptr.

So pass the Emperor and check within the function if he exists?

Reply:
yup.